editor/sidebar/post-status/index.js

```diff
@@ -27,9 +27,14 @@ class PostStatus extends Component {
 	}
 
 	render() {
-		const { status, onUpdateStatus, suggestedFormat } = this.props;
+		const { status, onUpdateStatus, suggestedFormat, post } = this.props;
 		const onToggle = () => {
-			const updatedStatus = status === 'pending' ? 'draft' : 'pending';
+			let updatedStatus;
+			if ( status !== 'pending' ) {
+				updatedStatus = 'pending';
+			} else {
+				updatedStatus = post.status && post.status !== 'pending' ? post.status : 'publish';
```

Comment:
1. Navigate to Gutenberg demo page
2. Click "Pending review" twice (toggle on, then toggle off)
3. Click "Save"

Expected: A draft is saved
Actual: The post is published